service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageDocumentsControllerPopulateListMidEventTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.CourtAdminDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.utils.TestDataHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.DocumentRouter.AMEND;
+import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.DocumentRouter.DELETE;
+import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.DocumentRouter.UPLOAD;
+import static uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement.EMPTY;
+import static uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement.builder;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(ManageDocumentsController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class ManageDocumentsControllerPopulateListMidEventTest extends AbstractControllerTest {
+
+    private static final DocumentReference DOCUMENT = TestDataHelper.testDocumentReference();
+
+    ManageDocumentsControllerPopulateListMidEventTest() {
+        super("manage-docs");
+    }
+
+    @Test
+    void shouldBuildDynamicListOfDocumentsWhenUserSelectsAmend() {
+        List<Element<CourtAdminDocument>> courtAdminDocuments = buildDocuments();
+
+        CaseData caseData = CaseData.builder()
+            .uploadDocumentsRouter(AMEND)
+            .otherCourtAdminDocuments(courtAdminDocuments)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "populate-list");
+        DynamicList documentList = mapper.convertValue(response.getData().get("courtDocumentList"), DynamicList.class);
+
+        DynamicList expectedList = dynamicList(courtAdminDocuments.get(0).getId(), courtAdminDocuments.get(1).getId());
+
+        assertThat(documentList).isEqualTo(expectedList);
+    }
+
+    //RDM-9147
+    @Test
+    void shouldRemoveEditedDocumentIfNoReplacementDocumentUploaded() {
```

Comment:
Isn't it tested in above file?

Reply:
true, I can delete since it is same test